trio_run_in_process/run_in_process.py

```diff
@@ -193,17 +194,40 @@ async def _open(
                         # If a keyboard interrupt is encountered relay it to the
                         # child process and then give it a moment to cleanup before
                         # re-raising
+                        logger.debug(
+                            "Got KeyboardInterrupt, sending SIGINT to %s", proc
+                        )
                         try:
                             proc.send_signal(signal.SIGINT)
-                            with trio.move_on_after(2):
-                                await proc.wait()
+                            try:
+                                with trio.fail_after(constants.SIGINT_TIMEOUT_SECONDS):
+                                    await proc.wait()
+                            except trio.TooSlowError:
```

Comment:
I find that I tend towards this exception handling pattern over the alternative of using `move_on_after(...) as scope` and then performing a `if scope.cancelled_caught` for the timeout code...

I would like to figure out at some point whether there is any benefit to one or the other.  Maybe the `try/except` has some additional performance overhead?  Maybe the `try/except` is more explicit about what is happening?  Maybe the `try/except` is more visually confusing because of the extra level of indentation....

Nothing actionable here, just saying some things outloud that are in my brain.

Reply:
Yeah, both have their drawbacks, but I tend to prefer the try/except